index.html

```diff
@@ -1642,6 +1642,18 @@ <h3>Selective Disclosure</h3>
 all of the credential's claims to be revealed.
 
           </li>
+          <li>
+<b> Hashed Values</b> - With this method, the <a>issuer</a> issue a single
+<a>verifiable credential</a> containing all the subject's <a>claims</a>.
+However, each <a>claim</a> value is created by hashing the actual value with
+a different nonce. The value that is issued for the <a>claim</a> then
+comprises an object containing the nonce and the hashed value of the nonce
+concatenated with the <a>claim</a> value. In addition, the
+<code>credentialSubject</code> contains the identifier of the hashing
+algorithm used to create all the hashed values. The holder includes the actual
+values of the <a>claims</a> that are to be revealed to the <a>verifier</a> in
+the <a>verifiable presentation</a>.
```

Comment:
More rewording is needed.  Perhaps replacing your "actual value" with "original 
literal value" would work... as --

```
However, each <a>claim</a> value in the <a>verifiable credential</a> is created
by hashing the original literal value with a different nonce. The value for the
<a>claim</a> in the <a>verifiable credential</a> then comprises an object
containing 1) the nonce and 2) the hashed value of the nonce concatenated with
the original literal <a>claim</a> value. In addition, the
<code>credentialSubject</code> contains the identifier of the hashing algorithm
used to create all the hashed values. The holder includes the original literal
values of the <a>claims</a> they intend to reveal to the <a>verifier</a> in their
<a>verifiable presentation</a>.
```

I am a bit confused by `the <code>credentialSubject</code> contains the identifier of the hashing algorithm`.  Does the `<code>credentialSubject</code>` here not contain the identifier of the subject of the credential?  Or does it include *both* identifiers (making it an array, rather than a single value)?  Or something else?

Reply:
I think that 'actual value' is better than 'original literal value'

The <code>credentialSubject</code> comprises a whole set of attributes, including the <code>id</code>. This is a proposal to add another property to this set i.e. the hashing algorithm. I will clarify this.

Reply:
Having a hashing algorithm as a property of a `credentialSubject` makes very little sense. It would mean, for example, that a VC about David Chadwick would state that David has a hashing algorithm of SHA-256.

It sounds like this information should be captured in the `proof` section somewhere instead.

Reply:
The hashing algorithm is actually a property of the claim value e.g. the claim {name=David} is transposed into {name={hashID=SHA-256, nonce=123....9, value=a8clyga}}. But since the same hashing algorithm is used for all claim values, there is no point in repeating the hashing algorithm ID multiple times. So we move it up a level to the encapsulating object, which is the credentialSubject, and put it there.

Reply:
Has this been resolved? I lean towards what @dlongley said above. The hashing algorithm, and all cryptographic attributes should be decoupled from the contents of the `credentialSubject`.

Reply:
If we had a data model in which a credentialSubject contained an ID and a set of claims, and each claim contained an ID and a set of properties, then we could have explicitly placed the hashID in the claim object. But unfortunately the data model has not explicitly identified the claim object with a claim ID and so the fact that this exists is obscured (and has led to many previous misunderstandings and issues.) Thus you think (wrongly) that the hashID is a property of the credentialSubject, when in fact, it is a property of the (invisible) claim object.